drivers/watchdog/wdt_dw.c

```diff
@@ -15,14 +15,38 @@
 #include "wdt_dw.h"
 #include "wdt_dw_common.h"
 
+/* Check if reset property is defined */
+#define WDT_DW_IS_INST_RESET_EN(inst)	DT_NODE_HAS_PROP(DT_DRV_INST(inst), resets)
+#define WDT_DW_CHECK_RESET(inst)	WDT_DW_IS_INST_RESET_EN(inst) ||
+#define WDT_DW_RESET_SUPPORT		(DT_INST_FOREACH_STATUS_OKAY(WDT_DW_CHECK_RESET) 0)
+
+#if WDT_DW_RESET_SUPPORT
+#include <zephyr/drivers/reset.h>
+#endif
+
+/* Check if clock manager is supported */
+#define WDT_DW_IS_INST_CLK_MGR_EN(inst)	DT_PHA_HAS_CELL(DT_DRV_INST(inst), clocks, clkid)
+#define WDT_DW_CHECK_CLK_MGR(inst)	WDT_DW_IS_INST_CLK_MGR_EN(inst) ||
+#define WDT_DW_CLK_MANAGER_SUPPORT	(DT_INST_FOREACH_STATUS_OKAY(WDT_DW_CHECK_CLK_MGR) 0)
```

Comment:
This really is macromagic and macrobatics

Reply:
I am just curious if it is necessary to use `#if xxx` "everywhere"? Do we really need to lower the footprint? Just too many macros, I think, some of them can be replaced by C code?

Reply:
I have updated the changes these with macro(DT_ANY_INST_HAS_PROP_STATUS_OKAY) defined in the devicetree.h, which is similar implementation